src/main/java/org/thoughtcrime/securesms/qr/QrCodeHandler.java

```diff
@@ -55,8 +55,13 @@ public void handleQrData(String rawString) {
                 showFingerPrintError(builder, nameAndAddress);
                 break;
 
-            case DcContext.DC_QR_FPR_OK:
             case DcContext.DC_QR_ADDR:
+                if (dcContext.isChatmail()) {
+                    dcContext.deleteContact(qrParsed.getId());
+                    DcHelper.showEncryptionRequiredDialog(activity, nameAndAddress);
+                    return;
+                }
```

Comment:
this is a bit aggressive, the contact might exist and have encryption, but we can't check if the contact exists since the QR parsing from core always creates a contact as side effect just by the action of scanning the QR which is a bit dangerous and questionable, probably should be fixed in core 

cc @link2xt, @iequidoo

Reply:
other option would be for now to allow to create chats with email address in QR it is not that common and the other changes catch the more common situations

Reply:
There are many ways to get contacts w/o keys for encryption anyway, e.g. contacts are auto-created from incoming messages' From/To/Cc. What happens now if you try to start a chat with such a contact? So maybe we should allow creating them, but handle this later? Though probably we should warn a user after scanning such a QR if we see that we don't have the contact's public key

Reply:
The contact may still be useful, you can copy its address and use your another account to send a message f.e.

Reply:
how can I check if a contact has encryption key or not?

Reply:
You can use jsonrpc's `CommandApi::get_contact_encryption_info()`. But the problem is that currently it returns the stock string "No encryption." instead of an empty string or other meaningful value for contacts w/o public keys. Also we have `ContactObejct::is_verified` flag, but don't have anything for the encryption state. Do i correctly understand that adding it to `ContactObject` fits, so the app can show a contact info after scanning a QR + a warning if we don't have the contact's key. I'm not sure it should be done this way, users sometimes don't read warnings, but not creating a contact at all may be worse.

Another way is not create a contact, but show an error with the scanned address so that the user can copy it at least

Reply:
so there is no way to know if a contact has encryption available, cheking for CommandApi::get_contact_encryption_info() string that is meant for display is no good, we need an api to check this, meanwhile we can just leave qr scanning mailto links as-is without the warning since we can't effectively take action with what core provides today

Reply:
> so there is no way to know if a contact has encryption available

Created a PR for this: https://github.com/deltachat/deltachat-core-rust/pull/5824, please check if it is helpful